src/System.Linq.Expressions/tests/MemberInit/BindTests.cs

```diff
@@ -223,21 +223,6 @@ public void StaticField(bool useInterpreter)
             Assert.Equal("ABC", PropertyAndFields.StaticStringField);
         }
 
-        [Theory, ClassData(typeof(CompilationTypes))]
-        public void StaticReadonlyField(bool useInterpreter)
-        {
-            MemberInfo member = typeof(PropertyAndFields).GetMember(nameof(PropertyAndFields.StaticReadonlyStringField))[0];
-            Expression<Func<PropertyAndFields>> assignToReadonly = Expression.Lambda<Func<PropertyAndFields>>(
-                Expression.MemberInit(
-                    Expression.New(typeof(PropertyAndFields)),
-                    Expression.Bind(member, Expression.Constant("ABC" + useInterpreter))
-                    )
-                );
-            Func<PropertyAndFields> func = assignToReadonly.Compile(useInterpreter);
-            func();
-            Assert.Equal("ABC" + useInterpreter, PropertyAndFields.StaticReadonlyStringField);
-        }
```

Comment:
I assume both the `false` and `true` variants of this test fail now (by design) with your change?

Reply:
Unfortunately, no ... the compiled version will still "succeed."

Reply:
There are number of other discrepancies between interpreted and compiled expressions if the expression is invalid. This is just one more.